src/main/kotlin/ua/marchenko/artauction/auth/jwt/JwtAuthenticationFilter.kt

```diff
@@ -1,52 +1,51 @@
 package ua.marchenko.artauction.auth.jwt
 
-import jakarta.servlet.FilterChain
-import jakarta.servlet.http.HttpServletRequest
-import jakarta.servlet.http.HttpServletResponse
-import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
-import org.springframework.security.core.context.SecurityContextHolder
-import org.springframework.security.core.userdetails.UserDetails
-import org.springframework.security.web.authentication.WebAuthenticationDetailsSource
 import org.springframework.stereotype.Component
-import org.springframework.web.filter.OncePerRequestFilter
 import ua.marchenko.artauction.auth.service.CustomUserDetailsServiceImpl
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.context.ReactiveSecurityContextHolder
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.web.server.ServerWebExchange
+import org.springframework.web.server.WebFilter
+import org.springframework.web.server.WebFilterChain
+import reactor.core.publisher.Mono
 
 @Component
 class JwtAuthenticationFilter(
     private val userDetailsService: CustomUserDetailsServiceImpl,
     private val jwtService: JwtService,
-) : OncePerRequestFilter() {
-
-    override fun doFilterInternal(
-        request: HttpServletRequest,
-        response: HttpServletResponse,
-        filterChain: FilterChain
-    ) {
-        val authHeader: String? = request.getHeader(HEADER_AUTHORIZATION)
+) : WebFilter {
+
+    @Suppress("ForbiddenVoid", "ReturnCount")
+    override fun filter(exchange: ServerWebExchange, chain: WebFilterChain): Mono<Void> {
```

Comment:
```suggestion
    @Suppress("ReturnCount")
    override fun filter(exchange: ServerWebExchange, chain: WebFilterChain): Mono<Unit> {
```
can it be fixed like that?